src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,294 @@
+        ennreal.zero_to_real, sub_zero],
+    exact ennreal.to_real_nonneg },
+end
+
+section
+
+variables {u v w : set α}
+
+/-- A subset `v` of a null-set `w` has zero measure if `w` is a subset of a positive set `u`. -/
+lemma subset_positive_null_set
+  (hu : measurable_set u) (hv : measurable_set v) (hw : measurable_set w)
+  (hsu : 0 ≤[u] s) (hw₁ : s w = 0) (hw₂ : w ⊆ u) (hwt : v ⊆ w) : s v = 0 :=
+begin
+  have : s v + s (w \ v) = 0,
+  { rw [← hw₁, ← of_union set.disjoint_diff hv (hw.diff hv),
+        set.union_diff_self, set.union_eq_self_of_subset_left hwt],
+    apply_instance },
+  have h₁ := nonneg_of_zero_le_restrict _ (restrict_le_restrict_subset _ _ hu hsu (hwt.trans hw₂)),
+  have h₂ := nonneg_of_zero_le_restrict _
+    (restrict_le_restrict_subset _ _ hu hsu ((w.diff_subset v).trans hw₂)),
+  linarith,
+end
+
+/-- A subset `v` of a null-set `w` has zero measure if `w` is a subset of a negative set `u`. -/
+lemma subset_negative_null_set
+  (hu : measurable_set u) (hv : measurable_set v) (hw : measurable_set w)
+  (hsu : s ≤[u] 0) (hw₁ : s w = 0) (hw₂ : w ⊆ u) (hwt : v ⊆ w) : s v = 0 :=
+begin
+  have : s v + s (w \ v) = 0,
+  { rw [← hw₁, ← of_union set.disjoint_diff hv (hw.diff hv),
+        set.union_diff_self, set.union_eq_self_of_subset_left hwt],
+    apply_instance },
+  have h₁ := nonpos_of_restrict_le_zero _ (restrict_le_restrict_subset _ _ hu hsu (hwt.trans hw₂)),
+  have h₂ := nonpos_of_restrict_le_zero _
+    (restrict_le_restrict_subset _ _ hu hsu ((w.diff_subset v).trans hw₂)),
+  linarith,
+end
+
+/-- If the symmetric difference of two positive sets is a null-set, then so are the differences
+between the two sets. -/
+lemma of_diff_eq_zero_of_symm_diff_eq_zero_positive
+  (hu : measurable_set u) (hv : measurable_set v)
+  (hsu : 0 ≤[u] s) (hsv : 0 ≤[v] s) (hs : s (u Δ v) = 0) :
+  s (u \ v) = 0 ∧ s (v \ u) = 0 :=
+begin
+  rw restrict_le_restrict_iff at hsu hsv,
+  have a := hsu (hu.diff hv) (u.diff_subset v),
+  have b := hsv (hv.diff hu) (v.diff_subset u),
+  erw [of_union (set.disjoint_of_subset_left (u.diff_subset v) set.disjoint_diff)
+        (hu.diff hv) (hv.diff hu)] at hs,
+  rw zero_apply at a b,
+  split,
+  all_goals { linarith <|> apply_instance <|> assumption },
+end
+
+/-- If the symmetric difference of two negative sets is a null-set, then so are the differences
+between the two sets. -/
+lemma of_diff_eq_zero_of_symm_diff_eq_zero_negative
+  (hu : measurable_set u) (hv : measurable_set v)
+  (hsu : s ≤[u] 0) (hsv : s ≤[v] 0) (hs : s (u Δ v) = 0) :
+  s (u \ v) = 0 ∧ s (v \ u) = 0 :=
+begin
+  rw restrict_le_restrict_iff at hsu hsv,
+  have a := hsu (hu.diff hv) (u.diff_subset v),
+  have b := hsv (hv.diff hu) (v.diff_subset u),
+  erw [of_union (set.disjoint_of_subset_left (u.diff_subset v) set.disjoint_diff)
+        (hu.diff hv) (hv.diff hu)] at hs,
+  rw zero_apply at a b,
+  split,
+  all_goals { linarith <|> apply_instance <|> assumption },
+end
+
+lemma of_diff_of_diff_eq_zero
```

Comment:
can you move this lemma to a different place (and state it for general vector measures)?